ec2/keypair.py

```diff
@@ -30,21 +31,45 @@
 def create(**kwargs):
     """ This will create the key pair within the region you are currently
         connected to.
+        Requires:
+            ctx.node.properties['resource_id']
+        Sets:
+            ctx.instance.runtime_properties['aws_resource_id']
+            ctx.instance.runtime_properties['key_path']
     """
+
     ec2_client = connection.EC2ConnectionClient().client()
-    key_pair_name = ctx.node.properties['name']
+
+    if ctx.node.properties['use_external_resource']:
+        key_pair_id = ctx.node.properties['resource_id']
+        key_pair = utils.get_key_pair_by_id(key_pair_id)
+        ctx.instance.runtime_properties['aws_resource_id'] = key_pair.name
+        key_path = get_key_file_path(ctx=ctx)
+        ctx.logger.info(key_path)
+        if not search_for_key_file(key_path):
+            raise NonRecoverableError('use_external_resource was specified, '
+                                      'and a name given, but the key pair was '
+                                      'not located on the filesystem.')
+        return
+
+    key_pair_name = ctx.node.properties['resource_id']
+
     ctx.logger.info('Creating key pair.')
```

Comment:
this should probably be `debug` since the user doesn't care that you're currently creating a keypair.. they care that you created it.

we should check that `debug` is viewable in the logs..